packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -374,9 +376,22 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     refreshLeadingPadding();
   }
 
+  @override
+  void dispose() {
+    _localTextEditingController?.dispose();
+    _localTextEditingController = null;
+    super.dispose();
+  }
+
   @override
   void didUpdateWidget(DropdownMenu<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
+    if (oldWidget.controller != widget.controller) {
```

Comment:
The old code wasn't handling the case where the widget got a new controller later on.  It would just ignore any controller that was set later, and continue using the one it originally constructed.